core/domain/classifier_services.py

```diff
@@ -162,58 +162,37 @@ def get_classifier_by_id(classifier_id):
     return classifier
 
 
-def _create_classifier(classifier):
+def create_classifier(job_id, classifier):
     """Creates classifier data model in the datastore given a classifier
        domain object.
 
     Args:
+        job_id: str. ID of the ClassifierTrainingJob corresponding to the
+            classifier.
         classifier: Domain object for the classifier.
 
     Returns:
         classifier_id: str. ID of the classifier.
     """
-    classifier_id = classifier_models.ClassifierDataModel.create(
-        classifier.id, classifier.exp_id,
-        classifier.exp_version_when_created,
-        classifier.state_name, classifier.algorithm_id,
-        classifier.classifier_data, classifier.data_schema_version)
-    return classifier_id
-
-
-def _update_classifier(classifier_data_model, state_name):
-    """Updates classifier data model in the datastore given a classifier
-    domain object.
-
-    Args:
-        classifier_data_model: Classifier data model instance in datastore.
-        state_name: The name of the state.
-
-    Note: All of the properties of a classifier are immutable,
-    except for state_name.
-    """
-    classifier_data_model.state_name = state_name
-    classifier_data_model.put()
-
-
-def save_classifier(classifier):
-    """Checks for the existence of the model.
-    If the model exists, it is updated using _update_classifier method.
-    If the model doesn't exist, it is created using _create_classifier method.
-
-    Args:
-        classifier: Domain object for the classifier.
-
-    Returns:
-        classifier_id: str. ID of the classifier.
-    """
-    classifier_id = classifier.id
     classifier_data_model = classifier_models.ClassifierDataModel.get(
-        classifier_id, False)
+        job_id, False)
```

Comment:
The second arg should be written as strict=False since there's already a default kwarg specified for 'strict'. Ditto elsewhere.